polaris-core/src/main/java/io/polaris/core/persistence/resolver/PolarisResolutionManifest.java

```diff
@@ -403,7 +403,11 @@ public PolarisResolvedPathWrapper getResolvedTopLevelEntity(
     if (resolvedCacheEntry == null) {
       return null;
     }
-    return new PolarisResolvedPathWrapper(
-        List.of(getResolvedRootContainerEntity(), new ResolvedPolarisEntity(resolvedCacheEntry)));
+
+    ResolvedPolarisEntity resolvedRootContainerEntity = getResolvedRootContainerEntity();
```

Comment:
Looks like there are more calls to `getResolvedRootContainerEntity()` that can fail with a direct or indirect NPE.
@collado-mike WDYT?